### ISSUE TITLE
`moment?` broken

### ISSUE THREAD

**Issue by sorpaas** [97jaz/gregor#20]
In the current release (racket 6.7), `(moment? (moment 1970))` returns `#f`.

**Comment by 97jaz** [comment]
Interesting. I checked on 6.6, and it works there. I should probably get 6.7 on this machine now. Let's see... Hmm, I just installed 6.7 (OS X, 64-bit), and it worked there too:

![screen shot 2016-12-24 at 1 25 25 am](https://cloud.githubusercontent.com/assets/984854/21465582/e6761758-c977-11e6-8b75-6c2613a8ee72.png)

What build are you using?

**Comment by sorpaas** [comment]
I'm using NixOS's Racket build (https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/racket/default.nix).

Turns out that I messed up Racket's phase system. I tried to create some really complicated macros, that have some code (using gregor) running in phase 1, and then pass that along to get compared in phase 0. At some stage in requiring all the modules, `#t` turned to `#f`. It never worked. I managed to solve the problem by reducing the number of macros I use.

How important it is to keep a good diet of macros...
